R/1.0_module_data_summary.R

```diff
@@ -0,0 +1,395 @@
+#'   useShinyjs(),
+#'   actionButton("hide_content", label = "hide", icon = icon("xmark")),
+#'   actionButton("show_content", label = "show", icon = icon("check")),
+#'   actionButton("toggle_content", label = "toggle", icon = icon("angle-down")),
+#'   tags$br(),
+#'   tags$div(
+#'     id = "content",
+#'     verbatimTextOutput("printout")
+#'   )
+#' )
+#'
+#' server <- function(input, output, session) {
+#'   observeEvent(input$hide_content,
+#'     {
+#'       hide("content")
+#'       toggle_icon("toggle_content", c("fa-angle-down", "fa-angle-right"), one_way = TRUE)
+#'     },
+#'     ignoreInit = TRUE
+#'   )
+#'
+#'   observeEvent(input$show_content,
+#'     {
+#'       show("content")
+#'       toggle_icon("toggle_content", c("fa-angle-right", "fa-angle-down"), one_way = TRUE)
+#'     },
+#'     ignoreInit = TRUE
+#'   )
+#'
+#'   observeEvent(input$toggle_content,
+#'     {
+#'       toggle("content")
+#'       toggle_icon("toggle_content", c("fa-angle-right", "fa-angle-down"))
+#'     },
+#'     ignoreInit = TRUE
+#'   )
+#'
+#'   output$printout <- renderPrint({
+#'     head(faithful, 10)
+#'   })
+#' }
+#' if (interactive()) {
+#'   shinyApp(ui, server)
+#' }
+#'
+#' @name toggle_button
+#' @rdname toggle_button
+#' @keywords internal
+toggle_icon <- function(input_id, icons, one_way = FALSE) {
+  checkmate::assert_string(input_id)
+  checkmate::assert_character(icons, len = 2L)
+  checkmate::assert_flag(one_way)
+
+  expr <-
+    if (one_way) {
+      sprintf(
+        "$('#%s i').removeClass('%s').addClass('%s');",
+        input_id, icons[1], icons[2]
+      )
+    } else {
+      sprintf("$('#%s i').toggleClass('%s');", input_id, paste(icons, collapse = " "))
+    }
+
+  shinyjs::runjs(expr)
+
+  invisible(NULL)
+}
+
+#' @rdname toggle_button
+#' @keywords internal
+toggle_title <- function(input_id, titles, one_way = FALSE) {
+  checkmate::assert_string(input_id)
+  checkmate::assert_character(titles, len = 2L)
+  checkmate::assert_flag(one_way)
+
+  expr <-
+    if (one_way) {
+      sprintf(
+        "$('a#%s').attr('title', '%s');",
+        input_id, titles[2]
+      )
+    } else {
+      sprintf(
+        paste0(
+          "var button_id = 'a#%1$s';",
+          "var curr = $(button_id).attr('title');",
+          "if (curr == '%2$s') { $(button_id).attr('title', '%3$s');",
+          "} else { $(button_id).attr('title', '%2$s');",
+          "}"
+        ),
+        input_id, titles[1], titles[2]
+      )
+    }
+
+  shinyjs::runjs(expr)
+
+  invisible(NULL)
+}
+
+#' @rdname module_data_summary
+#' @keywords internal
+get_filter_overview <- function(teal_data) {
+  logger::log_trace("srv_data_overiew-get_filter_overview initialized")
+  datanames <- teal.data::datanames(teal_data())
+  joinkeys <- teal.data::join_keys(teal_data())
+  filtered_data_objs <- sapply(datanames, function(name) teal_data()@env[[name]])
+  unfiltered_data_objs <- sapply(datanames, function(name) teal_data()@env[[paste0(name, "_raw")]])
```

Comment:
Bug here with IRIS dataset

```R
options(
  teal.log_level = "ERROR",
  teal.show_js_log = TRUE,
  # teal.bs_theme = bslib::bs_theme(version = 5),
  shiny.bookmarkStore = "server"
)

pkgload::load_all("../teal")

data <- teal_data() |>
  within(IRIS <- iris)

datanames(data) <- "IRIS"
init(
  data = data,
  modules = list(
    example_module()
  )
) |>
  shiny::runApp()
```

```suggestion
  filtered_data_objs <- sapply(datanames, function(name) teal_data()@env[[name]], simplify = FALSE)
  unfiltered_data_objs <- sapply(datanames, function(name) teal_data()@env[[paste0(name, "_raw")]], simplify = FALSE)
```

Reply:
Also, consider using `teal.code::get_env` instead of accessing the slot directly.

Reply:
@averissimo thanks. Tested just on one app that has multiple datasets and failed to catch this one. I should do more thourough tests

Reply:
Will merge this, and will push changes with `teal.code::get_env` in other commit